environment.cpp

```diff
@@ -127,15 +138,20 @@ double dummy_function(std::vector<double> input)
 
 void switch_env_function(environment &e)
 {
-  if(are_same_env_functions(e.get_current_function(), e.get_env_function_A())){
+  if(e.get_name_current_function()=='A'){
       e.change_env_function(e.get_env_function_B());
+      e.switch_name_current_function();
     }
-  else
+  else if (e.get_name_current_function()=='B'){
     e.change_env_function(e.get_env_function_A());
+    e.switch_name_current_function();
+  }
+  else throw;
```

Comment:
I am pretty sure there is a container somewhere in the standard library that would make our life much easier. 
About this we could also think of improving the design slightly by making use of the string -> function pointer `map` we have already. But this is for future us to take care.